src/t_hash.c

```diff
@@ -298,9 +882,13 @@ hashTypeIterator *hashTypeInitIterator(robj *subject) {
     hi->subject = subject;
     hi->encoding = subject->encoding;
 
-    if (hi->encoding == OBJ_ENCODING_LISTPACK) {
+    if (hi->encoding == OBJ_ENCODING_LISTPACK ||
+        hi->encoding == OBJ_ENCODING_LISTPACK_TTL)
+    {
         hi->fptr = NULL;
         hi->vptr = NULL;
+        hi->tptr = NULL;
+        hi->expire_time = EB_EXPIRE_TIME_INVALID;
     } else if (hi->encoding == OBJ_ENCODING_HT) {
         hi->di = dictGetIterator(subject->ptr);
```

Comment:
forget to initialize hi->expire_time to EB_EXPIRE_TIME_INVALID  when hash is hashtable?

Reply:
ht does not use this field but probably it is better to match behaviors. Maybe we should delete `expire_time` variable in the iterator and read value from `hi->tpr` in `hashTypeCurrentFromListpack()` similar to what is happening in `hashTypeCurrentFromHashTable()`.

Reply:
added a comment, hope it is clear